src/model_config_utils.cc

```diff
@@ -1768,6 +1779,12 @@ ValidateModelOutput(
         "shape tensors are only supported for TensorRT platform");
   }
 
+  if ((platform != kTensorRTPlanPlatform) && io.is_non_linear_format_io()) {
```

Comment:
We should also check for the number of dimension to be 3 or not for non_linear_format_io tensor.

I would advise to create a new function called `ValidateNonLinearFormatIO()` which could check the platflorm and number of dims. Then call to this function when a nonlinear tensor is encountered.